### ISSUE TITLE
Which models are supported？

### ISSUE THREAD

**Issue by 729973389** [browser-use/browser-use#12]
Hello, there is only
GPT-4o
GPT-4o Mini
Claude 3.5 Sonnet
LLama 3.1 405B
These four models have passed the test of this project, and the other LangChain chat models have not been tested, right? I used two models, Qianfan and Tongyi

My code is as follows:
from langchain_community.llms import QianfanLLMEndpoint
from langchain_community.llms import Tongyi
from browser_use import Agent

chatBot=Tongyi()
result=Tongyi().invoke("What NFL team won the Super Bowl in the year Justin Bieber was born?"
"请说中文")
print(result)

agent = Agent(
task="请你帮我打开这个网站:https://www.bilibili.com",
llm=chatBot,
)

# 运行 agent
result = await agent.run()

# 输出结果
print(result)

**Comment by 2br-2b** [comment]
In theory, any of these models should be supported: https://python.langchain.com/docs/integrations/chat/

**Comment by MagMueller** [comment]
Yes,  all from the list, which are able of tool-calling, structured-output and multimodal. 

For some models, the task is still quite hard and they tend to forget quickly your defined task.

**Comment by 729973389** [comment]
是的，作者说是这样的，但是我在使用的过程中发现，目前是调用openai是可以的，我只用tongyi是不行的，但是我用调用openai的api就可以

import asyncio
from langchain_openai import ChatOpenAI
from browser_use import Agent
# 创建 ChatOpenAI 客户端实例
chat = ChatOpenAI(
    api_key="",  # my tongyi API Key
    model="qwen-plus",  # model name
    base_url="https://dashscope.aliyuncs.com/compatible-mode/v1" 
)

agent = Agent(
    # task="Go to hackernews on show hn and give me top 10 post titles, their points and hours. Calculate for each the ratio of points per hour.",

    task="请你去百度中搜索 2024 年排名前 3 的 AI 公司（不要用谷歌），找出每家公司为他们的模型使用的具体硬件是什么。",
    llm=chat,

)

async def main():
    # Run agent
    try:
        result = await agent.run()
        # print(result)
        print("result-------->",result[0].text)
        # print("result-------->")
    except Exception as e:
        print("发生了一个错误，",e)
    # except :
    #     print("The structured output feature is not implemented in this model. Please use a different model.")

asyncio.run(main())

**Comment by 729973389** [comment]
Yes, the author said so, but I found in the process of using, at present, it is OK to call openai, I only use tongyi is not OK, but I can use the api to call openai

### Here is the code available:

import asyncio
from langchain_openai import ChatOpenAI
from browser_use import Agent
# 创建 ChatOpenAI 客户端实例
chat = ChatOpenAI(
    api_key="",  # my tongyi API Key
    model="qwen-plus",  # model name
    base_url="https://dashscope.aliyuncs.com/compatible-mode/v1" 
)

agent = Agent(
        task="请你去百度中搜索 2024 年排名前 3 的 AI 公司（不要用谷歌），找出每家公司为他们的模型使用的具体硬件是什么。",
    llm=chat,

)

async def main():
    # Run agent
    try:
        result = await agent.run()
        # print(result)
        print("result-------->",result[0].text)
        # print("result-------->")
    except Exception as e:
        print("发生了一个错误，",e)
    # except :
    #     print("The structured output feature is not implemented in this model. Please use a different model.")

asyncio.run(main())





### This is not available code:

import asyncio
from browser_use import Agent

from langchain_community.llms import Tongyi


chatBot=Tongyi()
result=Tongyi().invoke("What NFL team won the Super Bowl in the year Justin Bieber was born?"
                       "请说中文")
print(result)
# 创建 Agent 实例并设定任务

agent = Agent(
    # task="Go to hackernews on show hn and give me top 10 post titles, their points and hours. Calculate for each the ratio of points per hour.",

    task="Go to bilibili and give me top 10 post titles, their points and hours. Calculate for each the ratio of points per hour.",
    llm=chatBot,

)


async def main():
    # 运行 agent
    result =  await agent.run()

    # 输出结果
    print(result)

asyncio.run(main())

**Comment by izambe** [comment]
it works with openai model from github https://github.com/marketplace/models/azure-openai/gpt-4o-mini

**Comment by manugarri** [comment]
@izambe do you have a replicable snippet?

**Comment by izambe** [comment]
> @izambe do you have a replicable snippet?

Sure,

``
import os
from asyncio import run
from browser_use import Agent
from langchain_ollama import ChatOllama
from dotenv import load_dotenv
from langchain_openai import ChatOpenAI

load_dotenv()

llm = ChatOpenAI(
    base_url="https://models.inference.ai.azure.com",
    api_key=os.environ["GITHUB_TOKEN"],
    model="gpt-4o-mini"
)

agent = Agent(
    task="open https://www.google.com/ and search for jesus",
    llm=llm,
)

run(agent.run())
``

**Comment by TerminalDZ** [comment]
> > @izambe do you have a replicable snippet?
> 
> Sure,
> 
> ```
> import os
> from asyncio import run
> from browser_use import Agent
> from langchain_ollama import ChatOllama
> from dotenv import load_dotenv
> from langchain_openai import ChatOpenAI
> 
> load_dotenv()
> 
> llm = ChatOpenAI(
>     base_url="https://models.inference.ai.azure.com",
>     api_key=os.environ["GITHUB_TOKEN"],
>     model="gpt-4o-mini"
> )
> 
> agent = Agent(
>     task="open https://www.google.com/ and search for jesus",
>     llm=llm,
> )
> 
> run(agent.run())
> ```

**Error :**
`python .\agent.py
INFO     [browser_use] BrowserUse logging setup complete with level info
INFO     [root] Anonymized telemetry enabled. See https://github.com/gregpr07/browser-use for more information.
INFO     [agent] 🚀 Starting task: open https://www.google.com/ and search for jesus
INFO     [agent]
📍 Step 1
ERROR    [agent] ❌ Result failed 1/3 times:
 Error code: 400 - {'error': {'code': 'BadRequest', 'message': 'response_format value as json_schema is enabled only for api versions 2024-08-01-preview and later'}}
INFO     [agent]
📍 Step 1
ERROR    [agent] ❌ Result failed 2/3 times:
 Error code: 400 - {'error': {'code': 'BadRequest', 'message': 'response_format value as json_schema is enabled only for api versions 2024-08-01-preview and later'}}
INFO     [agent]
📍 Step 1
ERROR    [agent] ❌ Result failed 3/3 times:
 Error code: 400 - {'error': {'code': 'BadRequest', 'message': 'response_format value as json_schema is enabled only for api versions 2024-08-01-preview and later'}}
ERROR    [agent] ❌ Stopping due to 3 consecutive failures`

**Comment by TerminalDZ** [comment]
> it works with openai model from github https://github.com/marketplace/models/azure-openai/gpt-4o-mini

**Error_** :
`python .\agent.py INFO     [browser_use] BrowserUse logging setup complete with level info INFO     [root] Anonymized telemetry enabled. See https://github.com/gregpr07/browser-use for more information. INFO     [agent] 🚀 Starting task: open https://www.google.com/ and search for jesus INFO     [agent] 📍 Step 1 ERROR    [agent] ❌ Result failed 1/3 times: Error code: 400 - {'error': {'code': 'BadRequest', 'message': 'response_format value as json_schema is enabled only for api versions 2024-08-01-preview and later'}} INFO     [agent] 📍 Step 1 ERROR    [agent] ❌ Result failed 2/3 times: Error code: 400 - {'error': {'code': 'BadRequest', 'message': 'response_format value as json_schema is enabled only for api versions 2024-08-01-preview and later'}} INFO     [agent] 📍 Step 1 ERROR    [agent] ❌ Result failed 3/3 times: Error code: 400 - {'error': {'code': 'BadRequest', 'message': 'response_format value as json_schema is enabled only for api versions 2024-08-01-preview and later'}} ERROR    [agent] ❌ Stopping due to 3 consecutive failures  `

**Code:**
`import os
from asyncio import run
from browser_use import Agent
from langchain_ollama import ChatOllama
from dotenv import load_dotenv
from langchain_openai import ChatOpenAI

load_dotenv()

llm = ChatOpenAI(
    base_url="https://models.inference.ai.azure.com",
    api_key=os.environ["GITHUB_TOKEN"],
    model="gpt-4o-mini"

)

agent = Agent(
    task="open https://www.google.com/ and search for jesus",
    llm=llm,
)

run(agent.run())`

**Comment by TerminalDZ** [comment]
**Use:**
`import os
from asyncio import run
from browser_use import Agent
from langchain_ollama import ChatOllama
from dotenv import load_dotenv
from langchain_openai import ChatOpenAI
from langchain_openai import AzureChatOpenAI


load_dotenv()

llm = AzureChatOpenAI(
    azure_endpoint="https://models.inference.ai.azure.com",
    model="gpt-4o",
    api_version="2024-08-01-preview",
    api_key=os.environ["GITHUB_TOKEN"]
)

agent = Agent(
    task="""
    1. Navigate to https://algeria.blsspainvisa.com/
    2. Select location: Algiers
    3. Choose visa type: Tourist (First Time)
    4. Schedule appointment:
       a. Click "Schedule Appointment"
       b. Fill personal details: name, passport number, email, phone
       c. Select date and time
    5. Review and confirm appointment
    6. Note payment instructions if applicable
    7. Save/print appointment confirmation
    """,
    llm=llm,
)

run(agent.run())`
